src/main/resources/js/src/lib/visualisation.js

```diff
@@ -0,0 +1,144 @@
+import * as d3 from "d3";
+
+export function visualise(data) {
+	const pieDiv = d3.select('body').append('div').attr('class', 'toolTip');
+
+	const pieWidth = 500;
+	const pieHeight = 500;
+	const pieRadius = Math.min(pieWidth, pieHeight) / 2;
+
+	function getRandomColor() {
+		const letters = '0123456789ABCDEF';
+		let color = '#';
+		for (let i = 0; i < 6; i++) {
+			color += letters[Math.floor(Math.random() * 16)];
+		}
+		return color;
+	}
+
+	const arc = d3.arc()
+	.outerRadius(pieRadius - 10)
+	.innerRadius(pieRadius - 70);
+
+	const pie = d3.pie()
+	.sort(null)te
+	.value(d => d.value);
+
+	const svgPie = d3.select('body').append('svg')
+	.attr('width', pieWidth)
+	.attr('height', pieHeight)
+	.append('g')
+	.attr('transform', `translate(${pieWidth / 2},${pieHeight / 2})`);
+
+	const g = svgPie.selectAll('.arc')
+	.data(pie(data))
+	.enter().append('g')
+	.attr('class', 'arc');
+
+	g.append('path')
+	.attr('d', arc)
+	.style('fill', d => getRandomColor(d.data.label));
+
+	g.on('mousemove', d => {
+		pieDiv.style('left', `${d3.event.pageX+10}px`);
+		pieDiv.style('top', `${d3.event.pageY-25}px`);
+		pieDiv.style('display', 'inline-block');
+		pieDiv.html(`${d.data.label}<br>${d.value} vote${d.data.value > 1 ? 's' : ''}`);
+	});
+
+	g.on('mouseout', d => {
+		pieDiv.style('display', 'none');
+	});
+
+	g.append('text').attr('transform', (d) => {
+		console.log(d);
+		return `translate(${arc.centroid(d)})`;
+	})
+	.attr('dy', '.35em')
+	.text((d) => d.data.label)
+	.attr('class', 'value');
+
+	const div = d3.select('body').append('div').attr('class', 'toolTip');
+
+	const axisMargin = 20;
+	const margin = 40;
+	const valueMargin = 4;
+	let width = parseInt(d3.select('body').style('width'), 10);
+	let height = parseInt(d3.select('body').style('height'), 10);
+	const barHeight = (height - axisMargin - margin * 2) * 0.4 / data.length;
+	const barPadding = (height - axisMargin - margin * 2) * 0.4 / data.length;
+	let bar;
+	let svg;
+	let scale;
+	let xAxis;
+	let labelWidth = 0;
+	width *= .5;
+	height *= .8;
+	console.log(width);
+	const max = d3.max(data, d => d.value);
```

Comment:
Getting undefined somewhere with the `max` variable. Not sure if it's `d3` or the own variable